ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+                   self.dry_bulb_condition.modifier_schedule,
+                   self.humidity_condition.hum_type, '',
+                   self.humidity_condition.schedule, '', '',
+                   self.humidity_condition.wet_bulb_range,
+                   self.humidity_condition.barometric_pressure,
+                   self.wind_condition.wind_speed,
+                   self.wind_condition.wind_direction,
+                   self.wind_condition.rain,
+                   self.wind_condition.snow_on_ground,
+                   self.sky_condition.daylight_savings_indicator,
+                   self.sky_condition.solar_model,
+                   self.sky_condition.beam_shced,
+                   self.sky_condition.diff_sched, '', '', '']
+
+        # assign humidity values based on the type of criteria
+        if self.humidity_condition.hum_type == 'Wetbulb' or \
+                self.humidity_condition.hum_type == 'Dewpoint':
+                    ep_vals[9] = self.humidity_condition.hum_value
+        elif self.humidity_condition.hum_type == 'HumidityRatio':
+            ep_vals[11] = self.humidity_condition.hum_value
+        elif self.humidity_condition.hum_type == 'Enthalpy':
+            ep_vals[12] = self.humidity_condition.hum_value
+
+        # assign sky condition values based on the solar model
+        if self.sky_condition.solar_model == 'ASHRAEClearSky':
+            ep_vals[25] = self.sky_condition.clearness
+        if self.sky_condition.solar_model == 'ASHRAETau':
+            ep_vals[23] = self.sky_condition.tau_b
+            ep_vals[24] = self.sky_condition._tau_d
+            ep_vals.pop()
+
+        # put everything together into one string
+        ep_string = ' SizingPeriod:DesignDay,\n'
+        separator = ','
+        for i, val in enumerate(ep_vals):
+            if i == len(ep_vals) - 1:
+                separator = ';'
+            spacer = ' ' * (60 - len(str(val)))
+            ep_string = ep_string + '  ' + str(val) + separator + \
+                spacer + comments[i] + '\n'
+
+        return ep_string
+
+    @property
+    def name(self):
+        """Get or set the name."""
+        return self._name
+
+    @name.setter
+    def name(self, data):
+        assert isinstance(data, str), 'Design day name must be' \
+            ' text. Got {}'.format(type(data))
+        self._name = data
+
+    @property
+    def day_type(self):
+        """Get or set the type of design day."""
+        return self._day_type
+
+    @day_type.setter
+    def day_type(self, data):
+        assert data in day_types, 'day_type {} is not' \
+            ' recognized'.format(str(data))
+        self._day_type = data
+
+    @property
+    def location(self):
+        """Get or set the location."""
+        return self._location
+
+    @location.setter
+    def location(self, data):
+        assert hasattr(data, 'isLocation'), 'Expected' \
+            ' isLocation type. Got {}'.format(str(data))
```

Comment:
1. `type(data)` will be more informative than `str(data)`.
2. format does convert the type to string automatically.
3. Expected Location not isLocation.

Reply:
Sorry.  This was mostly a brain fart. I have gone back and made sure all of these are correct.